zerver/lib/webhooks/git.py

```diff
@@ -174,6 +174,7 @@ def get_pull_request_event_message(
     assignees: Optional[List[Dict[str, Any]]] = None,
     type: str = "PR",
     title: Optional[str] = None,
+    should_use_branch_info_template: Optional[bool] = None,
```

Comment:
Let's start using Python's feature of optional-only arguments so that this parameter can't be passed as a positional argument:
```suggestion
    title: Optional[str] = None,
    *,
    should_use_branch_info_template: Optional[bool] = None,
```

Really the same thing should be done for… well, really for *all* of this function's parameters, as I don't think it's transparent from the function's name what should be even the first or second parameter. That may be an annoying refactor; a good start would be to insert the `*,` anywhere in the list, as early as it can go without causing more than a handful of errors that need to be fixed up. But even without that, a good *first* step is just to draw a line in the sand at the existing level of messiness, by inserting it here before the newly-added parameter.

Reply:
I've inserted the `*,` as early as I can in the latest commit. I actually wonder if there is any resources to read about what it does.

Reply:
@Joelute See https://peps.python.org/pep-3102/

Reply:
Thank you!